### PR TITLE
MGDOBR-771: move internal validation to json schema

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-771](https://issues.redhat.com/browse/MGDOBR-771)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: manager/src/main/java/com/redhat/service/smartevents/manager/connectors/ConnectorsServiceImpl.java [hunk 1]
```diff
@@ -54,21 +57,21 @@ public void createConnectorEntity(Processor processor) {
...
     @Transactional(Transactional.TxType.MANDATORY)
     // Connector should always be marked for creation in the same transaction as a Processor
     public void createConnectorEntity(Processor processor, Source source) {
-        GatewayConnector<Source> sourceConnector = gatewayConfigurator.getSourceConnector(source.getType());
+        if (!processorCatalogService.isConnector(ProcessorType.SOURCE, source.getType())) {
+            return;
+        }
         String topicName = gatewayConfiguratorService.getConnectorTopicName(processor.getId());
-        persistConnectorEntity(processor, topicName, sourceConnector.getConnectorType(), sourceConnector.getConnectorTypeId(), sourceConnector.connectorPayload(source, topicName));
+        persistConnectorEntity(processor, topicName, ConnectorType.SOURCE, source.getType(), gatewayConnector.connectorPayload(source, topicName));
```
Comment (r00ta) [inline manager/src/main/java/com/redhat/service/smartevents/manager/connectors/ConnectorsServiceImpl.java:74]: as we take exactly the same parameters specified by managed connectors I implemented the logic in just one `GatewayConnector` -> we do not need anymore one class for every action/source

File: manager/src/test/java/com/redhat/service/smartevents/manager/api/user/ProcessorAPITest.java [hunk 2]
```diff
@@ -467,7 +467,7 @@ public void addProcessorWithEmptyChannelParameterToBridge() {
         action.setMapParameters(params);
 
         Response response = TestUtils.addProcessorToBridge(bridgeResponse.getId(), new ProcessorRequest("myProcessor", null, null, action));
-        assertThat(response.getStatusCode()).isEqualTo(400);
+        assertThat(response.getStatusCode()).isEqualTo(202);
```
Comment (r00ta) [inline manager/src/test/java/com/redhat/service/smartevents/manager/api/user/ProcessorAPITest.java:470]: This is actually a valid request according to the json schema provided by managed connectors

File: manager/src/test/java/com/redhat/service/smartevents/manager/api/user/ProcessorAPITest.java [hunk 3]
```diff
@@ -483,7 +483,7 @@ public void addProcessorWithEmptyWebhookURLParameterToBridge() {
...
```
Comment (r00ta) [inline manager/src/test/java/com/redhat/service/smartevents/manager/api/user/ProcessorAPITest.java:486]: same

File: manager/src/test/java/com/redhat/service/smartevents/manager/api/user/SchemaAPITest.java [hunk 4]
```diff
@@ -46,27 +42,8 @@ public void cleanUp() {
     @Test
     public void testAuthentication() {
         TestUtils.getProcessorsSchemaCatalog().then().statusCode(401);
-        TestUtils.getSourceProcessorsSchema("Slack").then().statusCode(401);
-        TestUtils.getActionProcessorsSchema("Slack").then().statusCode(401);
-    }
-
-    @Test
-    @TestSecurity(user = TestConstants.DEFAULT_CUSTOMER_ID)
-    public void testSchemasAreIncludedInCatalog() {
```
Comment (r00ta) [inline manager/src/test/java/com/redhat/service/smartevents/manager/api/user/SchemaAPITest.java:55]: moved to `ProcessorCatalogTest`

File: processor/configuration/src/main/java/com/redhat/service/smartevents/processor/AbstractGatewayValidator.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    public AbstractGatewayValidator() {
+    }
+
+    public AbstractGatewayValidator(ProcessorCatalogService processorCatalogService) {
+        this.processorCatalogService = processorCatalogService;
+    }
+
+    protected ValidationResult applyAdditionalValidations(T gateway) {
+        return ValidationResult.valid();
```
Comment (r00ta) [inline processor/configuration/src/main/java/com/redhat/service/smartevents/processor/AbstractGatewayValidator.java:21]: If are there any additional validations to be applied by an action/source, the processor validator has to override this method

**Review by r00ta — COMMENTED** [review]
For reviewers

File: processor/configuration/src/main/java/com/redhat/service/smartevents/processor/AbstractGatewayValidator.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    protected AbstractGatewayValidator(ProcessorCatalogService processorCatalogService) {
+        this.processorCatalogService = processorCatalogService;
+    }
+
+    protected ValidationResult applyAdditionalValidations(T gateway) {
+        return ValidationResult.valid();
+    }
+
+    public ValidationResult isValid(T gateway) {
```
[the remark this answers is not in the document]
Comment (r00ta) [reply processor/configuration/src/main/java/com/redhat/service/smartevents/processor/AbstractGatewayValidator.java:24]: done

File: processor/configuration/src/main/resources/schemas/actions/catalog.json [hunk 7]
```diff
@@ -0,0 +1,18 @@
+[
+  {
+    "name": "kafka_topic_sink_0.1",
+    "isConnector": false
+  },
+  {
+    "name": "send_to_bridge_sink_0.1",
```
[the remark this answers is not in the document]
Comment (r00ta) [reply processor/configuration/src/main/resources/schemas/actions/catalog.json:7]: I've renamed it. For the name and description see https://issues.redhat.com/browse/MGDOBR-791

**Review by danielezonca — APPROVED** [review]
LGTM

**Review by danielezonca — APPROVED** [review]

**Review by sutaakar — APPROVED** [review]
Looks good to me
Checked just the code, left the local testing on Srihari

**Review by lucamolteni — APPROVED** [review]

File: processor/configuration/src/test/java/com/redhat/service/smartevents/processor/sources/slack/SlackSourceValidatorTest.java [hunk 8]
```diff
@@ -29,45 +35,38 @@ protected String getSourceType() {
     @Test
     void isInvalidWithNoParameters() {
         Map<String, String> params = new HashMap<>();
-        assertValidationIsInvalid(sourceWith(params), SlackSourceValidator.INVALID_CHANNEL_MESSAGE);
+        assertValidationIsInvalid(sourceWith(params), "$.slack_channel: is missing but it is required and $.slack_token: is missing but it is required");
```
[the remark this answers is not in the document]
Comment (lucamolteni) [reply processor/configuration/src/test/java/com/redhat/service/smartevents/processor/sources/slack/SlackSourceValidatorTest.java:38]: https://issues.redhat.com/browse/MGDOBR-801

[the remark this answers is not in the document]
Comment (lucamolteni) [reply processor/configuration/src/test/java/com/redhat/service/smartevents/processor/sources/slack/SlackSourceValidatorTest.java:38]: https://issues.redhat.com/browse/MGDOBR-704
https://issues.redhat.com/browse/MGDOBR-705

File: processor/configuration/src/main/java/com/redhat/service/smartevents/processor/AbstractGatewayValidator.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    @Override
+    public final ValidationResult isValid(T gateway) {
+        if (gateway.getParameters() == null) {
+            return ValidationResult.invalid();
+        }
+
+        com.networknt.schema.ValidationResult result = processorCatalogService.validate(getType(), gateway.getProcessorType(), gateway.getParameters());
+        if (!result.getValidationMessages().isEmpty()) {
```
Comment (manstis) [inline processor/configuration/src/main/java/com/redhat/service/smartevents/processor/AbstractGatewayValidator.java:31]: Note to @manstis: I'll need to refactor this to return multiple discrete results!

**Pull request merged**